src/BlockActions.js

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {
+  getLabel,
+  getNestedBlockDefinition,
+  getSiblingsIds, isSimpleLayout,
+  triggerCustomEvent, triggerKeyboardEvent
+} from './processing/utils';
+import {duplicateBlock, hideBlock} from './actions';
+import {refType} from './types';
+
+
+@connect((state, props) => {
+  const {fieldId, blockId} = props;
+  const blockDefinition = getNestedBlockDefinition(state, fieldId, blockId);
+  const siblings = getSiblingsIds(state, fieldId, blockId);
+  return {
+    isSimpleLayout: isSimpleLayout(blockDefinition, state[fieldId].isMobile),
+    blockDefinition,
+    siblings,
+    icons: state[fieldId].icons,
+    index: siblings.indexOf(blockId),
+  };
+}, (dispatch, props) => {
+  const {fieldId, blockId} = props;
+  return bindActionCreators({
+    hideBlock: () => hideBlock(fieldId, blockId),
+    duplicateBlock: () => duplicateBlock(fieldId, blockId),
+  }, dispatch);
+})
+class BlockActions extends (React.Component) {
+  static propTypes = {
+    fieldId: PropTypes.string.isRequired,
+    blockId: PropTypes.string.isRequired,
+    sortableBlock: PropTypes.bool,
+    canDuplicate: PropTypes.bool,
+    dragHandleRef: refType,
+  };
+
+  static defaultProps = {
+    sortableBlock: true,
+    canDuplicate: true,
+  };
+
+  get moveUpIcon() {
+    const {icons} = this.props;
+    if ((icons === undefined) || (icons.moveUp === undefined)) {
+      return <i className="fas fa-chevron-up" />;
```

Comment:
For all icons,

- Should be marked as `aria-hidden="true"` so screen readers ignore them
- Should have a replacement label for screen-readers
- Should be done with SVG instead of an icon font